NitroxModel/Helper/NitroxUser.cs

```diff
@@ -10,230 +10,257 @@
-                }
+            // When running tests LanguageFiles is in same directory
+            if (execParentDir != null && Directory.Exists(Path.Combine(execParentDir.FullName, "LanguageFiles")))
+            {
+                return execParentDir.FullName;
+            }
 
-                return null;
-            },
-            () =>
+            // NitroxModel, NitroxServer and other assemblies are stored in Nitrox.Launcher/lib
+            if (execParentDir?.Parent != null && Directory.Exists(Path.Combine(execParentDir.Parent.FullName, "Resources", "LanguageFiles")))
             {
-                using ProcessEx proc = ProcessEx.GetFirstProcess("Nitrox.Launcher");
-                string executable = proc?.MainModule?.FileName;
-                return !string.IsNullOrWhiteSpace(executable) ? Path.GetDirectoryName(executable) : null;
+                return execParentDir.Parent.FullName;
             }
-        };
 
-        public static string AppDataPath { get; } = appDataPath ??= Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), "Nitrox");
+            return null;
+        },
+        () =>
+        {
+            using ProcessEx proc = ProcessEx.GetFirstProcess("Nitrox.Launcher");
+            string executable = proc?.MainModule?.FileName;
+            return !string.IsNullOrWhiteSpace(executable) ? Path.GetDirectoryName(executable) : null;
+        }
+    };
 
-        /// <summary>
-        ///     Tries to get the launcher path that was previously saved by other Nitrox code.
-        /// </summary>
-        public static string LauncherPath
+    public static string AppDataPath
+    {
+        get
         {
-            get
+            if (appDataPath != null)
             {
-                if (launcherPath != null)
-                {
-                    return launcherPath;
-                }
+                return appDataPath;
+            }
 
-                foreach (Func<string> retriever in launcherPathDataSources)
+            string applicationData = null;
+
+            // On linux Environment.SpecialFolder.ApplicationData returns the Windows version inside wine, this bypasses that behaviour
+            string homeInWineEnv = Environment.GetEnvironmentVariable("WINEHOMEDIR");
+            if (homeInWineEnv is { Length: > 4 })
+            {
+                string homeInWine = homeInWineEnv[4..]; // WINEHOMEDIR is prefixed with \??\
+                if (Directory.Exists(homeInWine))
                 {
-                    string path = retriever();
-                    if (!string.IsNullOrWhiteSpace(path) && Directory.Exists(path))
+                    applicationData = Path.Combine(homeInWine, ".config");
+
+                    if (!Directory.Exists(applicationData))
                     {
-                        return launcherPath = path;
+                        Directory.CreateDirectory(applicationData);
                     }
```

Comment:
`Directory.CreateDirectory` does nothing if directory already exists.
```suggestion
                    Directory.CreateDirectory(applicationData);
```